RCode/NEFI/partition_uncertainty_DayLengthQuad.R

```diff
@@ -0,0 +1,274 @@
+# IC = initial conditions from end of calibration period 
+# N_out = number of time steps to forecast 
+# tau_add = process error (defaults to zero) 
+# n = number of monte carlo sims 
+forecast_gloeo <- function(IC, beta1, beta2, beta3, beta4, DL=DL, N_out = N_out, tau_add = 0, n = Nmc){
+  out <- matrix(NA, n, N_out) # setting up output 
+  gloeo_prev <- IC
+  for(i in 1:N_out){
+    lambda <- beta1 + beta2*gloeo_prev + beta3*DL[i] + beta4*DL[i]^2
+    out[,i] <- rnorm(n,lambda,tau_add) 
+    gloeo_prev <- out[,i] # update IC 
+  }
+  return(out)
+}
+
+## deterministic predictions 
+DL = c(dat$forecast$daylength) #DL during forecasted period
+N_out = nrow(dat$forecast) # length of forecast time points 
+IC = out[,mus]
+betas = out[,grep("beta",colnames(out))]
+betas.mean <- apply(betas,2,mean)
+
+gloeo.det <- forecast_gloeo(IC = mean(IC[,N]), # last column of cal is the initial condition for forecast 
+                            N_out = N_out,
+                            beta1 = betas.mean[1],
+                            beta2 = betas.mean[2],
+                            beta3 = betas.mean[3],
+                            beta4 = betas.mean[4],
+                            DL = DL,
+                            tau_add = 0,
+                            n = 1)
+
+## Plot run
+plot.run()
+lines(forecast_time, exp(gloeo.det), col="purple", lwd=3)
+
+
+######## initial condition uncertainty #######
+# sample rows from previous analysis
+prow = sample.int(nrow(out),Nmc,replace=TRUE)
+
+gloeo.I <- forecast_gloeo(IC = IC[prow, N],
+                      N_out = N_out,
+                      tau_add = 0,
+                      beta1 = betas.mean[1],
+                      beta2 = betas.mean[2],
+                      beta3 = betas.mean[3],
+                      beta4 = betas.mean[4],
+                      DL = DL,
+                      n = Nmc)
+gloeo.I.ci = apply(exp(gloeo.I), 2, quantile, c(0.025,0.5,0.975))
+
+## Plot run
+plot.run()
+ecoforecastR::ciEnvelope(forecast_time, gloeo.I.ci[1,], gloeo.I.ci[3,], col = col.alpha(N.cols[1], trans))
+lines(forecast_time, gloeo.I.ci[2,], lwd=0.5)
+
+###### parameter uncertainty #######
+gloeo.IP <- forecast_gloeo(IC=IC[prow,N],  ## shouldn't I be holding IC constant tho??
+                      N_out = N_out,
+                      beta1 = betas[prow,1],
+                      beta2 = betas[prow,2],
+                      beta3 = betas[prow,3],
+                      beta4 = betas[prow,4],
+                      DL = DL,
+                      tau_add=0,
+                      n=Nmc)
+
+## Plot run
+plot.run()
+gloeo.IP.ci = apply(exp(gloeo.IP),2,quantile,c(0.025,0.5,0.975))
+ecoforecastR::ciEnvelope(forecast_time,gloeo.IP.ci[1,],gloeo.IP.ci[3,],col=col.alpha(N.cols[2],trans))
+ecoforecastR::ciEnvelope(forecast_time,gloeo.I.ci[1,],gloeo.I.ci[3,],col=col.alpha(N.cols[1],trans))
+lines(forecast_time,gloeo.I.ci[2,],lwd=0.5)
+
+###### driver uncertainty ########## 
+# we don't have this for exponential 
```

Comment:
I _think_ `day length` might be a driver, but likely has extremely low / zero uncertainty

Reply:
I was definitely considering DL as a driver (and air temperature in the other model too) but the trouble is I don't *think* we currently have a way to estimate driver uncertainty in our models? I know we will when we have to add in sub-models for missing data in, for example, water temperature, but we don't currently have priors/variance/anything like that developed for drivers...

Reply:
Right, we could look at the sensor manufacturer's estimate of accuracy as a starting point for air temp / water temp driver uncertainty.